.github/workflows/tests.yaml

```diff
@@ -0,0 +1,30 @@
+name: tests
+
+on:
+  push:
+    branches:
+      - main
+      - "release-*"
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+"
```

Comment:
We only want this to run on all pull requests targeting the main branch.

https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request